src/gt4py/next/program_processors/runners/dace/gtir_to_sdfg_primitives.py

```diff
@@ -599,22 +608,20 @@ def translate_tuple_get(
         raise ValueError(f"Invalid tuple expression {node}")
     # Now we remove the tuple fields that are not used, to avoid an SDFG validation
     # error because of isolated access nodes.
-    unused_arg_nodes = gtx_utils.flatten_nested_tuple(
+    unused_data_nodes = gtx_utils.flatten_nested_tuple(
         tuple(arg for i, arg in enumerate(data_nodes) if i != index)
     )
     # However, for temporary fields inside the tuple (non-globals and non-scalar
     # values, supposed to contain the result of some field operator) the gt4py
     # domain inference should have already set an empty domain, so the corresponding
     # `arg` is expected to be None and can be ignored.
-    assert all(
-        not arg.dc_node.desc(ctx.sdfg).transient or isinstance(arg.gt_type, ts.ScalarType)
-        for arg in unused_arg_nodes
-        if arg is not None
-    )
-    unused_arg_nodes = tuple(arg for arg in unused_arg_nodes if arg is not None)
-    ctx.state.remove_nodes_from(
-        [arg.dc_node for arg in unused_arg_nodes if ctx.state.degree(arg.dc_node) == 0]
-    )
+    access_nodes_to_remove = [
+        arg.dc_node
+        for arg in unused_data_nodes
+        if arg is not None and not arg.dc_node.desc(ctx.sdfg).transient
```

Comment:
```suggestion
        if arg is None or arg.dc_node.desc(ctx.sdfg).transient
```
Should be equivalent and easier to read or do I miss something?